sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -173,6 +180,11 @@ abstract class BinaryArithmetic extends BinaryOperator with NullIntolerant {
   def calendarIntervalMethod: String =
     sys.error("BinaryArithmetics must override either calendarIntervalMethod or genCode")
 
+  /** Name of the function for this expression on [[DayTimeIntervalType]] and
+   *  [[YearMonthIntervalType]] types. */
+  def intervalMethod: String =
```

Comment:
Shall we reuse `exactMathMethod`?

Reply:
Sure, since `exactMathMethod` is always defined in the expressions, we can directly invoke `exactMathMethod.get`.

Reply:
we can use `assert` to make sure it's defined for interval types.